core/spec/models/spree/order_updater_spec.rb

```diff
@@ -238,33 +232,23 @@ module Spree
       end
 
       it "updates each shipment" do
-        shipment = stub_model(Spree::Shipment, order: order)
-        shipments = [shipment]
-        allow(order).to receive_messages shipments: shipments
-        allow(shipments).to receive_messages states: []
-        allow(shipments).to receive_messages ready: []
-        allow(shipments).to receive_messages pending: []
-        allow(shipments).to receive_messages shipped: []
+        create(:shipment, order: order)
 
-        expect(shipment).to receive(:update!).with(order)
+        expect_any_instance_of(Spree::Shipment).to receive(:update!)
```

Comment:
i don't think this really tests what the example says it does anymore. the example says that each shipment gets updated, but this just asserts than any shipment somewhere gets updated. same for the specs below